src/composables/RoutingSlip/useCreateRoutingSlipPayment.ts

```diff
@@ -1,20 +1,39 @@
-import { ref } from '@vue/composition-api'
+import { computed, onMounted, ref, watch } from '@vue/composition-api'
+
+import { createNamespacedHelpers } from 'vuex-composition-helpers'
+
+const routingSlipModule = createNamespacedHelpers('routingSlip') // specific module name
+const { useState, useMutations } = routingSlipModule
 
 // Composable function to inject Props, options and values to CreateRoutingSlipDetails component
 export function useCreateRoutingSlipPayment () {
-  // default the payment type of routing slip to cheque
-  const isPaymentCheque = ref<boolean>(true)
+  const isPaymentCheque = ref<boolean>(undefined)
   const createRoutingSlipChequePaymentRef = ref<HTMLFormElement>(null)
   const createRoutingSlipCashPaymentRef = ref<HTMLFormElement>(null)
 
+  // vuex state and mutations
+  const { isPaymentMethodCheque } = useState(['isPaymentMethodCheque'])
+  const { setIsPaymentMethodCheque } = useMutations(['setIsPaymentMethodCheque'])
+
+  // watch any changes and update to store
+  watch(isPaymentCheque, () => {
+    setIsPaymentMethodCheque(isPaymentCheque.value)
```

Comment:
check old value and new value before pushing to store or even we can use computed property here

Reply:
since it is a radio button, the value would be unique.